sdk/core/azure-core-http-netty/src/main/java/com/azure/core/http/netty/implementation/NettyAsyncHttpResponseBase.java

```diff
@@ -23,6 +24,7 @@ public abstract class NettyAsyncHttpResponseBase extends HttpResponse {
     //private NettyToAzureCoreHttpHeadersWrapper headers;
     private final HttpHeaders headers;
 
+    @SuppressWarnings("deprecation")
```

Comment:
is there something we can do to avoid suppressing deprecations where string name usage is valid? 

E.g. maybe we need to keep one not-deprecated set/add method for HttpClients that performs all necessary normalization ?

Reply:
sure, but maybe we can have a package=private method and accessor for valid usages or something? does not need to happen in this PR of course